packages/components/buttons/icon-button/src/icon-button.styles.ts

```diff
@@ -175,43 +60,91 @@ const getSizeStyles = (size: TIconButtonProps['size']) => {
   switch (size) {
     case 'small':
       return css`
-        height: ${buttonSizes.small};
-        width: ${buttonSizes.small};
+        height: ${designTokens.heightForButtonAsIconAsSmall};
+        width: ${designTokens.heightForButtonAsIconAsSmall};
       `;
     case 'medium':
       return css`
-        height: ${buttonSizes.medium};
-        width: ${buttonSizes.medium};
+        height: ${designTokens.heightForButtonAsIconAsMedium};
+        width: ${designTokens.heightForButtonAsIconAsMedium};
       `;
     case 'big':
       return css`
-        height: ${buttonSizes.big};
-        width: ${buttonSizes.big};
+        height: ${designTokens.heightForButtonAsIconAsBig};
+        width: ${designTokens.heightForButtonAsIconAsBig};
       `;
     default:
       return css``;
   }
 };
-const getThemeStyles = (theme: TIconButtonProps['theme']) => {
-  if (!theme) return css``;
 
-  if (theme === 'default') return css``;
+const getBaseStyles = (
+  theme: TIconButtonProps['theme'],
+  isDisabled: TIconButtonProps['isDisabled'],
+  isActive: boolean
+) => {
+  if (isDisabled) {
+    return css`
+      &,
+      &:hover {
+        background-color: ${designTokens.backgroundColorForButtonWhenDisabled};
+        border-color: ${designTokens.borderColorForButtonAsIconWhenDisabled};
+        color: ${designTokens.colorNeutral60};
+        box-shadow: none;
+      }
+    `;
+  }
+
+  if (!theme || theme === 'default') {
+    return css`
+      &:hover {
+        background-color: ${designTokens.backgroundColorForButtonWhenHovered};
+        box-shadow: ${designTokens.shadowForButtonWhenHovered};
+      }
+      ${isActive ? '&,' : ''}
+      &:active {
+        background-color: ${designTokens.backgroundColorForButtonWhenActive};
+        box-shadow: ${designTokens.shadowForButtonWhenActive};
+        border-color: ${designTokens.borderColorForButtonAsIcon};
+      }
+    `;
+  }
 
   switch (theme) {
     case 'primary':
       return css`
         &:hover {
-          background-color: ${designTokens.colorPrimary};
+          background-color: ${designTokens.backgroundColorForButtonAsIconAsPrimaryWhenHovered};
+          box-shadow: ${designTokens.shadowForButtonWhenHovered};
+        }
+        ${isActive ? '&,' : ''}
+        &:active {
+          background-color: ${designTokens.backgroundColorForButtonAsIconAsPrimaryWhenActive};
+          box-shadow: ${designTokens.shadowForButtonWhenActive};
+        }
+        ${isActive ? '&,' : ''}
+        &:hover,
+        &:active {
           border-color: ${designTokens.colorPrimary};
-          color: ${designTokens.colorSurface};
+          border-color: ${designTokens.borderColorForButtonAsIconAsPrimary};
```

Comment:
duplicated `border-color` rule

Reply:
Thanks for noticing 👍 

Updated in this commit: ef1c6f88e890539eb7dba2918d557d38ccbad782